### PR TITLE
API response (HAL+JSON)

### PR DESCRIPTION
**Pull request by kfangw**
### Github Issue
#118 

### Background
To enhance user(client-side) experience, HAL+JSON response is needed
Spec is here https://tools.ietf.org/html/rfc4627

### Solution
To make it easy, a library is adopted https://github.com/nvellon/hal
By using the library, the only thing needed is to implement `Mapper` interface
```
type Mapper interface {
	GetMap() Entry
}
```

### Possible Drawbacks
* prev/next for pagination is not yet implemented.
* The API response is still returned as-is form. I need to consider the clients.

### CODE REVIEW

**Comment by kfangw** [comment]
@spikeekips @anarcher @outsideris Please review this change.

@spikeekips Please double-check the keys of a response map such as `account_id`, `checkpoint`, etc..

**Comment by anarcher** [comment]
What about their responses has their own package? e.g. `/api/resources` ?

**Comment by anarcher** [comment]
The Resoures can be a implemantion of `common.Serializable`?  For Render, common interface for rendering can be useful. IMHO.  :-)

File: lib/api_resource.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+}
+
+type APIResourceOperation struct {
+	hash    string
+	txHash  string
+	funder  string //Source Account
+	account string //Target Account
+	otype   string
+	amount  string
+}
```
Comment (anarcher) [inline lib/api_resource.go]: IMHO, How about `resource.Operation` ? 
And  the `resource.Operation` has `block.Operation` like below:

```
type  Operation sturct {
      op *block.Operation
...
}
func (o Operation) GetMap() hal.Entry {
    return hal.Entry{
       "id" : o.op.hash,
      ...
   }
}
```
It makes senes? :-)

Comment (kfangw) [reply lib/api_resource.go]: I thought about that. But, I decided that the Resources does not know the blocks.

**Comment by kfangw** [comment]
I will update commit for your recommendation. Thx ~! @anarcher

**Comment by spikeekips** [comment]
@kfangw Like`funder` And `account` we’d better to use the existing terms, I think. 

- funder -> source
- account -> target

File: lib/api_resource.go (hunk 1 above)
Comment (anarcher) [reply lib/api_resource.go]: But Copy values can cost. humm. And binding blocks and resources can make handlers itself to complex or verbose IMHO. 

``` 
resource.NewAccount(block.BlockAccount) *resource.Account 
api.NewAPIResourceAccunt(block.BlockAccount) *api.ResourceAccount
```

Comment (kfangw) [reply lib/api_resource.go]: It may possible when the restructuring is done.
Let me change~

**Pull request merged**